### PR TITLE
allowed prometheus to gather metrics

### LINKED ISSUE #45: Allow prometheus to gather metrics published by controller-runtime
The context can be found here: https://github.com/kubernetes-sigs/jobset/issues/53
For leadeworkerset, we need the same support.

We should follow similar PR: https://github.com/kubernetes-sigs/jobset/pull/99

### PR DESCRIPTION
**Pull request by Edwinhr716**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
Allows prometheus to gather metrics

#### Which issue(s) this PR fixes

Fixes #45

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,9 +75,9 @@ help: ## Display this help.
 .PHONY: manifests
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) \
-		rbac:roleName=manager-role output:rbac:artifacts:config=config/rbac \
-		crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/crd/bases \
-		webhook output:webhook:artifacts:config=config/webhook \
+		rbac:roleName=manager-role output:rbac:artifacts:config=config/components/rbac \
```
Comment (liurupeng) [inline Makefile:78]: qq, @Edwinhr716  why do we need to move these files to /components/

Comment (Edwinhr716) [reply Makefile]: good question, was following what was done by Kueue https://github.com/kubernetes-sigs/kueue and JobSet https://github.com/kubernetes-sigs/jobset, but turns out the whole reorganization isn't needed for it to work. Fixed it

**Comment by ahg-g** [comment]
do we need to upgrade golang to 1.22? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_lws/68/pull-lws-test-integration-main/1772314340867706880

**Comment by liurupeng** [comment]
> do we need to upgrade golang to 1.22? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_lws/68/pull-lws-test-integration-main/1772314340867706880

@ahg-g yep, Edwin will migrate the go version in the PR as well to fix the issue

**Comment by ahg-g** [comment]
probable sending a separate PR is better.

**Comment by kannon92** [comment]
Yea, I just did this for jobset/kueue. Most likely you will need a test-infra PR first (as defaults seem to pull 1.21 right now).

And then I'd suggest verifying it on the presubmit.

**Comment by ahg-g** [comment]
@Edwinhr716 did you test it in practice?

**Comment by Edwinhr716** [comment]
Yes, I installed kube-prometheus in my cluster, and was able to query for the `controller_runtime_reconcile_errors_total` metric 
```
$ curl 'http://localhost:9090/api/v1/query?query=controller_runtime_reconcile_errors_total'
{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"controller_runtime_reconcile_errors_total","container":"kube-rbac-proxy","controller":"cert-rotator","endpoint":"https","instance":"10.60.9.6:8443","job":"lws-controller-manager-metrics-service","namespace":"lws-system","pod":"lws-controller-manager-66b9bbddcb-7n8xc","service":"lws-controller-manager-metrics-service"},"value":[1711470007.443,"0"]},{"metric":{"__name__":"controller_runtime_reconcile_errors_total","container":"kube-rbac-proxy","controller":"leaderworkerset","endpoint":"https","instance":"10.60.9.6:8443","job":"lws-controller-manager-metrics-service","namespace":"lws-system","pod":"lws-controller-manager-66b9bbddcb-7n8xc","service":"lws-controller-manager-metrics-service"},"value":[1711470007.443,"0"]},{"metric":{"__name__":"controller_runtime_reconcile_errors_total","container":"kube-rbac-proxy","controller":"pod","endpoint":"https","instance":"10.60.9.6:8443","job":"lws-controller-manager-metrics-service","namespace":"lws-system","pod":"lws-controller-manager-66b9bbddcb-7n8xc","service":"lws-controller-manager-metrics-service"},"value":[1711470007.443,"0"]}]}}
```

**Pull request merged**